sklearn/discriminant_analysis.py

```diff
@@ -254,6 +255,18 @@ def __init__(self, solver='svd', shrinkage=None, priors=None,
         self.store_covariance = store_covariance  # used only in svd solver
         self.tol = tol  # used only in svd solver
 
+    @property
+    def covariance_(self):
+        if hasattr(self, '_covariance_') and not self.store_covariance:
```

Comment:
Thanks. What you mean by "the former behavior" is always store `covariance_` and raise a warning in `_solve_svd`? just like:
 ```
         if not self.store_covariance:
            warnings.warn("from version 0.21 'covariance_' will be stored only"
                          " if 'store_covariance' is True", DeprecationWarning)
        self.covariance_ = _class_cov(X, y, self.priors_, shrinkage)
```